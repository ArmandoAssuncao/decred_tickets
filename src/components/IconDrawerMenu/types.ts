export interface IProps {
    color?: string
    onPress?: () => void
    size?: number
}

export interface IState {
}
