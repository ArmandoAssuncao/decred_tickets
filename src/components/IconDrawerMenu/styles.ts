import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        padding: '1rem',
    } as ViewStyle,
})
