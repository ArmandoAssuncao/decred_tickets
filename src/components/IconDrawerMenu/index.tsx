import React from 'react'
import Icon from 'react-native-vector-icons/FontAwesome'

import { IProps, IState } from './types'
import styles from './styles'

export default class IconDrawerMenu extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
        size: 25,
    }

    render(): JSX.Element {
        const { color, onPress, size } = this.props
        return (
            <Icon
                name="bars"
                size={size}
                color={color}
                onPress={onPress}
                style={styles.container}
            />
        )
    }
}
