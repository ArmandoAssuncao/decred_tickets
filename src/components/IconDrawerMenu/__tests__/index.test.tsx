import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import IconDrawerMenu from '../index'

describe('IconDrawerMenu', () => {
    const wrapper = shallow(<IconDrawerMenu onPress={jest.fn()} />)

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('has 1 Icon component', () => {
        expect(wrapper.find('Icon')).toHaveLength(1)
    })

    describe('actions', () => {
        it('call onPress when pressed', () => {
            wrapper.props().onPress()
            expect(wrapper.find('Icon').prop('onPress')).toBeCalled()
        })
    })
})
