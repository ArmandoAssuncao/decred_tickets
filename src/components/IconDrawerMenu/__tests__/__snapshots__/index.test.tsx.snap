// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconDrawerMenu match snapshot 1`] = `
<Icon
  allowFontScaling={false}
  name="bars"
  onPress={[MockFunction]}
  size={25}
  style={
    Object {
      "padding": 16,
    }
  }
/>
`;
